### DISCUSSION TITLE
Get original item in v-data-table(-server) slot?

### DISCUSSION THREAD

**Discussion by jacob-carlborg-apoex** [vuetifyjs/vuetify#17028] [Q&A] [answered]
For a `v-data-table` and `v-data-table-server`, what's the proper way to get the original item when using the [[`item.${string}`]](https://vuetifyjs.com/en/api/v-data-table/#slots-[%60item.{string}%60]) slot? This documentation [1] uses `item.raw`. The next example [2] uses `item.value`. This example [3] (which uses a different slot) uses `item.columns`. According to the API documentation, the `item` that is passed is of type [`InternalDataTableItem`](https://github.com/vuetifyjs/vuetify/blob/29777628ac8839c0548e869d3d350ed9fdcbb149/packages/vuetify/src/labs/VDataTable/types.ts#L43-L43). This type then breaks down to [`InternalItem`](https://github.com/vuetifyjs/vuetify/blob/29777628ac8839c0548e869d3d350ed9fdcbb149/packages/vuetify/src/composables/items.ts#L9-L19), but there's not documentation of what the members of the interface represent.

I put a breakpoint to see what is actually passed to the slot. Here's a screenshot of what's passed for a `v-data-table`:

<img width="377" alt="data_table" src="https://user-images.githubusercontent.com/82817095/228457450-7cfef843-123d-460d-a032-6f42fd006cf1.png">

I can see that the data is duplicated in five places: `item.columns`, `item.props.title`, `item.props.value`, `item.raw` and `item.value`.

For a `v-data-table-server` it looks like this:

<img width="463" alt="data_table_server" src="https://user-images.githubusercontent.com/82817095/228458005-787819ed-7052-41b7-bc5a-ae504530454d.png">

In this case the data is duplicated in three places: `item.columns`, `item.props.title`, `item.props.value` and `item.raw`. I have no idea what `item.value` and `item.props.value` are. I don't recognize that value (`21545`).

So what's the proper way to get the original item that, ideally, works for both `v-data-table` and `v-data-table-server`? I don't want to accidentally rely on internal implementation details.

Here's a snippet of my code:

```vue
<template>
  <v-card>
    <v-card-title>{{ t('title') }}</v-card-title>
    <v-card-text>
      <v-data-table-server
        v-model:page="page"
        v-model:items-per-page="pageSize"
        :headers="headers"
        :items="items"
        :items-length="total"
        @update:options="fetch"
      >
        <template #item.transaction_date="{item}">
          {{
            (() => {
              debugger
            })()
          }}
          {{ truncateDate(item.raw.transaction_date) }}
        </template>
      </v-data-table-server>
    </v-card-text>
  </v-card>
</template>
```

[1] https://vuetifyjs.com/en/components/data-tables/basics/#item
[2] https://vuetifyjs.com/en/components/data-tables/basics/#simple-checkbox
[3] https://vuetifyjs.com/en/components/data-tables/slots/#item-slot

**Comment by KaelWD** [comment, marked as answer]
InternalItem needs to be split up into separate interfaces for selects and tables, `item.title` and `item.props` don't really make sense for tables. The original item will always be under `item.raw` though.

**Reply by jacob-carlborg-apoex** [reply]
> The original item will always be under item.raw though.

Thanks, that was what I was looking for.
